src/ray/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.2)
+set(CMAKE_CXX_FLAGS "-g -Wall -Werror -std=c++11")
```

Comment:
we're currently using `-Wextra` (though we whitelist some things)

from `src/common/cmake/Common.cmake`

```
set(CMAKE_C_FLAGS "-g -Wall -Wextra -Werror=implicit-function-declaration -Wno-sign-compare -Wno-unused-parameter -Wno-type-limits -Wno-missing-field-initializers --std=c99 -D_XOPEN_SOURCE=500 -D_POSIX_C_SOURCE=200809L -fPIC -std=c99")
```

Reply:
This is currently not done for C++ files, I filed an issue: https://github.com/ray-project/ray/issues/1314

Let's do it as a followup PR.